docs/data-migration.md

```diff
@@ -0,0 +1,76 @@
+# Data migration
+
+This document describes the process of migrating `LND`'s database state from one
+type of database backend (for example the `bbolt` based database files `*.db`
+such as the `channel.db` or `wallet.db` files) to another (for example the new `postgres` or `sqlite` databases introduced in `lnd v0.14.0-beta`).
+
+**Note:** Currently only migrations from `bolt` to either `postgres` or `sqlite` are supported. KV based databases will be phased out in the future. We are planning to add additonal support to migrate all `etcd` databases to `sql` databases. Moreover it is currently also not supported to move the database from `sqlite` to `postgres` or vice versa, because this migration only takes care of the key value data and there already exist the possiblity to have the invoices in native sql which this tool does not support.
+
+
+## Prepare the destination database
+
+### Using postgres as the destination remote database
+
+Prepare a user and database as described in the [Postgres](postgres.md)
+documentation. You'll need the Data Source Name (DSN) for both the data
+migration and then the `lnd` configuration, so keep that string somewhere
+(should be something with the format of `postgres://xx:yy@localhost:5432/zz`).
+
+No additional steps are required to prepare the Postgres database for the data
+migration. The migration tool will create the database schema automatically, so
+no DDL scripts need to be run in advance.
+
+## Prepare the source database
+
+Assuming we want to migrate the database state from the pre-0.19.0 individual
+`bbolt` based `*.db` files to a remote database, we first need to make sure the
+source files are in the correct state.
+
+The following steps should be performed *before* running the data migration:
+1. Stop `lnd`
+2. Upgrade the `lnd` binary to the latest version (e.g. `v0.19.0-beta` or later)
+3. Make sure to add config options like `gc-canceled-invoices-on-startup=true`
```

Comment:
I think we should probably also mention `--db.prune-revocation` which will remove a huge chunk of data from the database.